mpas_analysis/shared/mpas_xarray/mpas_xarray.py

```diff
@@ -69,70 +152,216 @@ def subset_variables(ds, vlist):  # {{{
     # drop spurious coordinates
     ds = ds.drop(dropcoords)
 
-    assert len(ds.variables.keys()) > 0, \
-        'MPAS_XARRAY ERROR: Empty dataset is returned.\n' \
-        'Variables {}\nare not found within the dataset ' \
-        'variables: {}.'.format(vlist, allvars)
+    if len(ds.data_vars.keys()) == 0:
+        raise ValueError(
+                'Empty dataset is returned.\n'
+                'Variables {}\n'
+                'are not found within the dataset '
+                'variables: {}.'.format(variable_list, allvars))
+
+    return ds  # }}}
+
+
+def preprocess(ds, time_variable_name, variable_list, sel_values,
```

Comment:
This code has been largely unchanged, but it is highlighted as though it were new code.  In the future, please try to keep edits to code in-place so that we have a clearer idea of how the code evolves from commit to commit, which is obviously very important for debugging.